### PR TITLE
Refactor some of the methods to use default behaviours and tweaked if ordering

### PR DESCRIPTION
**Pull request by ryanm101**
Re-factored some of the statements to simplify (imo) their readability and reduce their complexity.

Change nested `if` statements to be `if and` as python tests the statements in order. 

E.g.
This code will always return making the `else` redundant  and we are testing for a negative which imo adds confusion as most people default to positive checks mentally. 
```python
if not self.read_meter1:
    return True
else:
    return self.read_modbus_data_meter("m1_", 40190)
```
we can change this to the following which, if the `if` statement evaluates false will return by default. 
```python
if self.read_meter1:
  return self.read_modbus_data_meter("m1_", 40190)
return True    
```

### CODE REVIEW

**Comment by binsentsu** [comment]
Hi, thanks for this. Can you fix conflict, so it merges?

**Comment by ryanm101** [comment]
Done

File: custom_components/solaredge_modbus/__init__.py [hunk 1]
```diff
@@ -829,59 +824,59 @@ def decode_string(decoder):
...
+        #0x7a - 4 - cumulative charged (Wh)
+        cumulative_charged = decoder.decode_64bit_uint()
+        #0x7E - 2 - current max size Wh
+        battery_max = decoder.decode_32bit_float()
+        #0x80 - 2 - available size Wh
+        battery_availbable = decoder.decode_32bit_float()
+        #0x82 - 2 - SoH %
+        battery_SoH = decoder.decode_32bit_float()
+        #0x84 - 2 - SoC %
+        battery_SoC = validate(decoder.decode_32bit_float(), ">", 0.0)
```
Comment (ryanm101) [inline custom_components/solaredge_modbus/__init__.py:855]: I do think this should be `battery_SoC = validate(decoder.decode_32bit_float(), ">=", 0.0)` as i'd have thought `0.0` to be valid.

File: custom_components/solaredge_modbus/__init__.py [hunk 2]
```diff
@@ -253,438 +253,435 @@ def read_modbus_data(self):
...
+        )
+        self.data[meter_prefix + "importvarhq4c"] = round(
+            importvarhq4c, abs(energyvarsf)
+        )
+
+        return True
 
     def read_modbus_data_inverter(self):
         inverter_data = self.read_holding_registers(unit=self._address, address=40071, count=38)
         if not inverter_data.isError():
```
Comment (binsentsu) [inline custom_components/solaredge_modbus/__init__.py:568]: think this 'not' need to be removed now according to switched if

**Review by binsentsu — CHANGES_REQUESTED** [review]

File: custom_components/solaredge_modbus/__init__.py (hunk 2 above)
Comment (ryanm101) [reply custom_components/solaredge_modbus/__init__.py:568]: Good Catch. Must have crept back in when i was remerging from master

**Review by binsentsu — APPROVED** [review]

**Pull request merged**